test/chan.js

```diff
@@ -0,0 +1,75 @@
+var nano = require('..');
+var test = require('tape');
+
+test('adding and removing subscription channels', function (t) {
+
+  t.plan(7);
+
+  var pub = nano.socket('pub');
+
+  //register some channels: hello, foo, bar
+  var sub = nano.socket('sub', {
+    chan:'hello,foo,bar' //use an array if prefixes w/ commas are needed
+  });
+
+  pub.bind('inproc://filter');
+  sub.connect('inproc://filter');
+
+  //the main object we'll use to establish channel registration and removal
+  var msgs = {
+    m0:'hello world',
+    m1:'hello world',
+    m2:'hello world',
+    m3:'bar world',
+    m4:'foo world'
+  }
```

Comment:
we should add a message that is not a chan we're subscribed too and explicitly fail if that message is encountered.